src/wallet/wallet.cpp

```diff
@@ -2861,26 +2861,36 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
     }
 
     //if we're doing only denominated, we need to round up to the nearest smallest denomination
-    if(nCoinType == ONLY_DENOMINATED) {
+    if (nCoinType == ONLY_DENOMINATED) {
         std::vector<CAmount> vecPrivateSendDenominations = CPrivateSend::GetStandardDenominations();
-        CAmount nSmallestDenom = vecPrivateSendDenominations.back();
-        // Make outputs by looping through denominations, from large to small
-        for (const auto& nDenom : vecPrivateSendDenominations)
-        {
-            for (const auto& out : vCoins)
-            {
-                //make sure it's the denom we're looking for, round the amount up to smallest denom
-                if(out.tx->tx->vout[out.i].nValue == nDenom && nValueRet + nDenom < nTargetValue + nSmallestDenom) {
-                    COutPoint outpoint = COutPoint(out.tx->GetHash(),out.i);
-                    int nRounds = GetCappedOutpointPrivateSendRounds(outpoint);
-                    // make sure it's actually anonymized
-                    if(nRounds < privateSendClient.nPrivateSendRounds) continue;
-                    nValueRet += nDenom;
-                    setCoinsRet.insert(std::make_pair(out.tx, out.i));
+        std::vector<CAmount> vecDenominationsAsAFee = vecPrivateSendDenominations;
+        std::reverse(vecDenominationsAsAFee.begin(), vecDenominationsAsAFee.end());
+
+        // Try to fit into fee that matches one of denominations, from small to large
+        for (const auto& nDenomAsAFee : vecDenominationsAsAFee) {
+            CAmount nMaxPSFee = std::min(nDenomAsAFee, maxTxFee);
```

Comment:
does validation.h#L61 need to be changed (ie change default max)?
current
```
//! -maxtxfee default
static const CAmount DEFAULT_TRANSACTION_MAXFEE = 0.2 * COIN; // "smallest denom" + X * "denom tails"
```

Reply:
Hmmm.. we should've probably reverted #737 (i.e. changed this to `0.1 * COIN` which is bitcoin's default) the last time we introduced smaller denom becasue the logic in original PR was no longer relevant even at that moment 🙈This probably deserves its own PR however, since it's not directly connected to changes introduced in this specific PR imo.

Reply:
see https://github.com/dashpay/dash/pull/2362